### PR TITLE
[Prim] simplify bn vjp code

### PR DESCRIPTION
**Pull request by cyber-pioneer**
### PR types
Others

### PR changes
Others

### Describe
Pcard-66969
Simplify bn vjp code and fix composite rule branch check.

### CODE REVIEW

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h [hunk 1]
```diff
@@ -1284,8 +1284,11 @@ void batch_norm_grad(const Tensor& x,
     case DataLayout::kNCHW: {
       auto nhwc_x = transpose<T>(x_data, nchw_to_nhwc_dim);
       auto nhwc_out_grad = transpose<T>(out_grad_data, nchw_to_nhwc_dim);
+      auto nhwc_out_grad_sum = sum<T>(nhwc_out_grad, reduce_axis, dtype, false);
 
       auto x_sub_mean = nhwc_x - mean_data;
+      auto nhwc_out_grad_mul_x_sub_mean_sum =
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: change name？

**Review by JiabinYang — COMMENTED** [review]
give some mem and ips value

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h (hunk 1 above)
Comment (cyber-pioneer) [reply paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: done

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**